### PR TITLE
Add some reasons for why binary mode might be used

### PR DESCRIPTION
**Pull request by duglin**
/cc @afrittoli @xibz

Signed-off-by: Doug Davis <dug@microsoft.com>

### CODE REVIEW

File: cloudevents/spec.md [hunk 1]
```diff
@@ -130,7 +130,15 @@ A "structured-mode message" is one where the event is fully encoded using
...
-body, and event attributes are stored as part of message meta-data.
+body, and event attributes are stored as part of message metadata.
+
+Often, binary mode is used when the producer of the CloudEvent wishes to add
+the CloudEvents metadata to an existing event without impacting the message's
+body. This means that in most cases a binary CloudEvents enabled message would
+not break an existing receiver's processing of the event because the message's
+metadata typically allows for extension attributes. In other words, a binary
+formatted CloudEvent would work for both a CloudEvents enabled receiver as well
+as one that is unaware of CloudEvents.
```
Comment (afrittoli) [inline cloudevents/spec.md:141]: Thanks @duglin - this is exactly the use case I had in mind!

File: cloudevents/spec.md [hunk 2]
```diff
@@ -130,7 +130,15 @@ A "structured-mode message" is one where the event is fully encoded using
 a stand-alone event format and stored in the message body.
 
 A "binary-mode message" is one where the event data is stored in the message
-body, and event attributes are stored as part of message meta-data.
+body, and event attributes are stored as part of message metadata.
+
+Often, binary mode is used when the producer of the CloudEvent wishes to add
+the CloudEvents metadata to an existing event without impacting the message's
```
Comment (lance) [inline cloudevents/spec.md:136]:
```suggestion
the CloudEvent's metadata to an existing event without impacting the message
```

File: cloudevents/spec.md [hunk 3]
```diff
@@ -130,7 +130,15 @@ A "structured-mode message" is one where the event is fully encoded using
...
+body. This means that in most cases a binary CloudEvents enabled message would
```
Comment (lance) [inline cloudevents/spec.md:137]:
```suggestion
body. In most cases a CloudEvent encoded as a binary-mode message will
```
I guess it's not just typos - I can't help wordsmithing sometimes - take it or not.

**Review by xibz — APPROVED** [review]
This is perfect! I really like the mention of not wanting to change the previous body's structure :)

File: cloudevents/spec.md (hunk 2 above)
Comment (duglin) [reply cloudevents/spec.md:136]: Done

File: cloudevents/spec.md (hunk 3 above)
Comment (duglin) [reply cloudevents/spec.md:137]: done!

**Comment by duglin** [comment]
thanks @lance 
Approved on the 7/21 call

**Pull request merged**